apps/hash-frontend/src/shared/layout/layout-with-header/notifications-dropdown.tsx

```diff
@@ -1,27 +1,27 @@
 import { faBell } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@hashintel/design-system";
-import { Box, ListItemText, Menu, Typography, useTheme } from "@mui/material";
 import {
-  bindMenu,
-  bindTrigger,
-  usePopupState,
-} from "material-ui-popup-state/hooks";
-import { FunctionComponent, useState } from "react";
+  Box,
+  svgIconClasses,
+  Typography,
+  typographyClasses,
+  useTheme,
+} from "@mui/material";
+import { FunctionComponent } from "react";
 
-import { MenuItem } from "../../ui";
+import { useNotifications } from "../../notifications-context";
+import { Link } from "../../ui";
 import { HeaderIconButton } from "./shared/header-icon-button";
 
 export const NotificationsDropdown: FunctionComponent = () => {
   const theme = useTheme();
-  const [notificationsLength] = useState(3);
-  const hasNotifications = !!notificationsLength;
-  const popupState = usePopupState({
-    variant: "popover",
-    popupId: "notifications-dropdown-menu",
-  });
+
+  const { notifications } = useNotifications();
+
+  const hasNotifications = !!notifications?.length;
 
   return (
-    <Box>
+    <Link noLinkStyle href="/inbox">
       <HeaderIconButton
```

Comment:
Spacing of this looks a little off... it looks like there is a bit more spacing between the bell and number
<img width="106" alt="Screenshot 2023-11-02 at 14 15 02" src="https://github.com/hashintel/hash/assets/37743469/79ec4f85-1b5c-4c9d-90f0-be2bab04596d">

vs implementation
<img width="65" alt="Screenshot 2023-11-02 at 14 16 22" src="https://github.com/hashintel/hash/assets/37743469/3544d398-553a-453f-8632-c2e4a2ac7423">

Also Y padding possibly a bit off, with the notification bell (and avatar) shorter than the create icon – we should make consistent
![Screenshot 2023-11-02 at 14 20 29](https://github.com/hashintel/hash/assets/37743469/ac42644e-7363-4c6c-a0c3-6f1ffe050bde)

Reply:
Have addressed in https://github.com/hashintel/hash/pull/3511/commits/8a49d84288c68bf9ca3aedbc630ed46b21906c28, now looks like this:

<img width="187" alt="image" src="https://github.com/hashintel/hash/assets/42802102/f56774d6-a946-4d18-af89-3c4d8a2a2dce">